ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Rasf/RasfParser.c

```diff
@@ -0,0 +1,103 @@
+/** @file
+  RASF table parser
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Reference(s):
+    - ACPI 6.5 Specification - August 2022
+**/
+
+#include <Library/PrintLib.h>
+#include <Library/BaseLib.h>
+#include <Library/UefiLib.h>
+#include "AcpiParser.h"
+#include "AcpiView.h"
+
+STATIC ACPI_DESCRIPTION_HEADER_INFO  AcpiHdrInfo;
+
+/**
+  This function is intended to dump out the PCC Id in the RASF Table
+
+  @param [in] Ptr     Pointer to the start of the PCC Id buffer
+  @param [in] Length  The length of the buffer (should be 12)
+**/
+STATIC
+VOID
+DumpPCCId (
+  IN UINT8   *Ptr,
+  IN UINT32  Length
+  )
+{
+  UINT32  Byte;
+
+  PrintFieldName (2, L"PCC Identifier");
+  for (Byte = 0; Byte < Length; Byte++) {
+    Print (L"%02X", Ptr[Byte]);
+    if (Byte < (Length - 1)) {
+      Print (L" ");
+    } else {
+      Print (L"\n");
+    }
+  }
+}
+
+/**
+  An ACPI_PARSER array describing the ACPI RASF Table.
+**/
+STATIC CONST ACPI_PARSER  RasfParser[] = {
+  PARSE_ACPI_HEADER (&AcpiHdrInfo),
+};
+
+/**
+  This function parses the ACPI RASF table.
+  When trace is enabled this function parses the RASF table and
+  traces the ACPI table fields.
+
+  This function also performs validation of the ACPI table fields.
+
+  @param [in] Trace              If TRUE, trace the ACPI fields.
+  @param [in] Ptr                Pointer to the start of the buffer.
+  @param [in] AcpiTableLength    Length of the ACPI table.
+  @param [in] AcpiTableRevision  Revision of the ACPI table.
+**/
+VOID
+EFIAPI
+ParseAcpiRasf (
+  IN BOOLEAN  Trace,
+  IN UINT8    *Ptr,
+  IN UINT32   AcpiTableLength,
+  IN UINT8    AcpiTableRevision
+  )
+{
+  UINT32  Offset;
+
+  if (!Trace) {
+    return;
+  }
+
+  // Parse ACPI Header + RASF "fixed" fields
+  Offset = ParseAcpi (
+             Trace,
+             0,
+             "RASF",
+             Ptr,
+             AcpiTableLength,
+             PARSER_PARAMS (RasfParser)
+             );
+
+  // Dump next 12 bytes as that's the ACPI spec definition of the PCC ID
+  if (Offset == (AcpiTableLength - 12)) {
+    DumpPCCId (
+      Ptr + Offset,
+      12
+      );
+  } else {
+    // Rest of ACPI table is not the right size
+    IncrementErrorCount ();
+    Print (
+      L"\nERROR: PCC Id should be 12 bytesm but is %d bytes",

```

Comment:
typo: `bytesm`

Reply:
Gee - how did that pass by? typo indeed. Will fix.